R/utils.R

```diff
@@ -74,6 +74,27 @@ check_bbi_args <- function(.args) {
     }
   }
   #If parallel is passed
+  else if(is.null(.args$parallel) && is.null(.args$threads))
```

Comment:
I'm getting pretty turned around looking at this long chain of conditions.

```R
  #If parallel not passed
  if (is.null(.args$parallel) && !is.null(.args$threads))
  {
    if (.args$threads > 1)
    {
      cmd_args <- c(cmd_args, "--parallel")
    }
  }
  #If parallel is passed
  else if(is.null(.args$parallel) && is.null(.args$threads))
  {
    .args$parallel <- FALSE
    cmd_args <- c(cmd_args, "--threads=1")

  }

  else if(!is.null(.args$parallel) && is.null(.args$threads))
  {
    .args$parallel <- FALSE
    cmd_args <- c(cmd_args, "--threads=1")

  }

  else if(is.null(.args$parallel) && is.null(.args$threads))
  {
    .args$parallel <- FALSE
    cmd_args <- c(cmd_args, "--threads=1")

  }

  else{
    if (.args$threads > 1 &&
        isFALSE(.args$parallel))
      warning("`threads > 1` but model will not run in parallel because `parallel = FALSE`")
  }
```

 *  The "#If parallel is passed" comment is stale, I guess, because the condition right below it now includes `is.null(.args$parallel)`.

 * I'm surprised to see `args$parallel` being set to false.  That doesn't align with the desired behavior mentioned in https://github.com/metrumresearchgroup/bbr/issues/385#issuecomment-1182171875

---

Based on reading the desired behavior in <https://github.com/metrumresearchgroup/bbr/issues/385#issuecomment-1182171875>, it seems like the only time we should be doing anything is when threads is passed an above 1, so perhaps it makes sense to use that as the top condition?  Maybe the whole block above could be reduced to


```R
  if (!is.null(.args$threads) && .args$threads > 1) {
    if (is.null(.args$parallel)) {
      cmd_args <- c(cmd_args, "--parallel")
    } else if (isFALSE(.args$parallel)) {
      warning("`threads > 1` but model will not run in parallel because `parallel = FALSE`")
    }
  }
```

?

With that, a submit_models test would fail because it wouldn't match the `--threads=1` that you added in this PR's 4ce574d8, but I don't understand the reason for that change.

Reply:
I think this makes sense. I was trying to think through all the combinations of what could be passed or not between parallel being set or not and threads.